indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskQueue.java

```diff
@@ -671,7 +672,7 @@ private void notifyStatus(final Task task, final TaskStatus taskStatus, String r
     // Save status to metadata store first, so if we crash while doing the rest of the shutdown, our successor
     // remembers that this task has completed.
     try {
-      final Optional<TaskStatus> previousStatus = taskStorage.getStatus(task.getId());
+      final Optional<TaskStatus> previousStatus = getTaskStatus(task.getId());
```

Comment:
This really should use the metadata store. The code block is only called when a task completes, and we need to check to make sure the metadata store has the correct status stored.

Reply:
Thanks, added a comment as well